### PR TITLE
[11.x.x ]Fixed XR camera FOV can be changed through inspector UI

### PR DESCRIPTION
**Pull request by thomas-zeng**
### Checklist for PR maker
- [ ] Have you added a backport label (if needed)? For example, the `need-backport-*` label. After you backport the PR, the label changes to `backported-*`.
- [ ] Have you updated the changelog? Each package has a `CHANGELOG.md` file.
- [ ] Have you updated or added the documentation for your PR? When you add a new feature, change a property name, or change the behavior of a feature, it's best practice to include related documentation changes in the same PR. If you do add documentation, make sure to add the relevant Graphics Docs team member as a reviewer of the PR. If you are not sure which person to add, see the [Docs team contacts sheet](https://docs.google.com/spreadsheets/d/1rgUWWgwLFEHIQ3Rz-LnK6PAKmbM49DZZ9al4hvnztOo/edit#gid=1058860420).
- [ ] Have you added a graphic test for your PR (if needed)? When you add a new feature, or discover a bug that tests don't cover, please add a graphic test.

---
### Purpose of this PR
- Fixed camera fov UI is not locked in XR.

---
### Testing status
- ABV
- QA

---
### Comments to reviewers
- Several features in unity uses camera fov for culling/lod selection and more. This PR added code to sync up XR culling fov with camera fov.
- To QA:
   -- I have tested with cube scene and template scene on SPI PC mockHMD, everything seems to work fine.
   -- The change could potentially impact culling/lod/cascaded shadow.

### CODE REVIEW

**Comment by DennisDeRykeUnity** [comment]
So far I have tested on Quest Link using: 
- Unity (trunk 7771e0d3dc48) 
- URP (universal/xr/fix-tweakable-fov c0ca087) 
- Test scenes crafted to exercise LOD, shadow cascades, and view frustum culling 

I tested each scene in Play mode with both MultiPass and Single Pass Instancing.  I tested each scene as a Standalone built player with one stereo mode or the other.  I also experimented in the Editor with non-XR behaviour for comparison, tried different cascade settings, and used the Occlusion Culling Visualization in Scene view.  

None of this revealed new bugs introduced by this PR.  However, the highest resolution LOD fails to appear in XR mode, even if I put my face right up to the geometry so that it started near-plane clipping.  This is not a problem that is introduced by this PR, because I see the same thing using URP 8.1.  I will file a Fogbugz case later. 

Prior to approving I intend to run my three test scenes on Quest (non-Link mode).

**Comment by DennisDeRykeUnity** [comment]
On Quest (non-Link mode) I tested using:
- Unity (trunk 7771e0d3dc48)
- URP (universal/xr/fix-tweakable-fov c0ca087)
- Test scenes crafted to exercise LOD, shadow cascades, and view frustum culling

Collectively these tests touched all four combinations of MultiView/MultiPass + Vulkan/OpenGLES3, though I did not run all four of these configurations with all thee of the test scenes.  

I saw no problems introduced by this PR.

File: com.unity.render-pipelines.universal/Runtime/UniversalRenderPipeline.cs [hunk 1]
```diff
@@ -302,6 +302,11 @@ static bool TryGetCullingParameters(CameraData cameraData, out ScriptableCulling
             if (cameraData.xr.enabled)
             {
                 cullingParams = cameraData.xr.cullingParams;
+
+                // Sync the FOV on the camera to match the projection from the XR device
+                if (!cameraData.camera.usePhysicalProperties)
```
Comment (phi-lira) [inline com.unity.render-pipelines.universal/Runtime/UniversalRenderPipeline.cs:307]: Do you think we might need to do this path in other places? If so maybe add an internal utility function to do it?

**Review by phi-lira — APPROVED** [review]

**Review by DennisDeRykeUnity — APPROVED** [review]
Based on my testing described in previous comments, I approve this PR contingent on its ABV becoming as green as recent Graphics master ABV.

**Comment by thomas-zeng** [comment]
I run Yamato ABV agasint custom revision 21.1 5181ec54d595526303eb2714f480cbc3ee1911bd
Some HDRP tests failed.
VFX_URP on Win_DX11_playmode_mono_apiNet2_Linear failed with error:
```
Overall status: FAIL
Reason(s): FAILURE REASON: One or more non-test related errors or failures occurred.
Total tests: 47
Failed test count: 0
Successful tests count: 47
Not run tests count: 0
```
Other URP tests passed.

**Review by phi-lira — APPROVED** [review]

**Comment by phi-lira** [comment]
Failure unrelated to this PR.

**Pull request merged**